### DISCUSSION TITLE
Backup Utility

### DISCUSSION THREAD

**Discussion by bierlingm** [lightningrodlabs/acorn#129] [Ideas]
We're looking to adopt Acorn for Clutter development and would want a way to continuously save the state of the project either locally or perhaps directly to a GitHub repo.

Would this be as simple as running a script that looks at the contents of a folder where the source chain is stored? Or would it need to execute the export function in the hApp itself?

This will become important for us, and I suspect for others, especially when it's an "open" project.

Otherwise, it would be amazing to have some way to distinguish participants in a project between editing-capable members and those not enabled.

cc @artbrock

**Comment by Connoropolous** [comment]
interesting. First question is:
1. do you mean something built in to Acorn? in that case, it could only run while Acorn is running. Or do you mean instead...
2. regardless of whether Acorn is running, a script runs on a known schedule creating a backup. 

Number 2 would suggest a script customized to someone's system, and would be more of a devops-y task

**Reply by bierlingm** [reply]
I'd welcome either solution, but don't want to put more burden on you guys to develop something.

@artbrock, @jost-s, I guess it would be fairly simple to just run a local script to back up the data available locally?

**Reply by jost-s** [reply]
@bierlingm Yes, we'd need to identify the data to back up and zip and copy it somewhere.

@Connoropolous Is project data stored in the sourcechain?

**Reply by Connoropolous** [reply]
@bierlingm mentioned in discord that they've managed to get an 'Automator' script for Mac running. 

I want to mention something that I find interesting though, which is that since 'dynamic arc sizing' and other things that make 'sharding' of the data enabled are now on by default, no node necessarily has a full sync database slice of the DHT. 

This suggests that using an actual zome call and queries that pull from the DHT might be the more effective way to secure the fullest possible data backups. 

In theory, if Acorn were running, a script could call the same zome calls that the UI does to fetch that data, and write it in a file to disk.

**Reply by jost-s** [reply]
In that case it's not trivial to do these backups, not like just backing up a folder from the file system.

@Connoropolous A project in Acorn has a UID for its DNA, if I understand correctly. Sharding doesn't kick in for less than ~100 nodes, so for a Clutter project it wouldn't happen.

**Reply by Connoropolous** [reply]
Oh ok, I didn't know the specifics, is it documented somewhere? In that case a local file folder backup would suffice

**Reply by bierlingm** [reply]
(The workflow currently just takes the "mac: ~/Library/Application Support/acorn" folder and copies its contents to a target folder)

**Reply by jost-s** [reply]
@Connoropolous Answer from Michael:

> hmm not really, this is probably the closest thing to documentation that I can think of: https://github.com/holochain/holochain/blob/develop/crates/kitsune_p2p/dht_arc/src/defaults.rs#L1-L11
